scripts/transfer_leads.py

```diff
@@ -0,0 +1,227 @@
+target_custom_fieldnames = [x['name'] for x in resp['data']]
+
+new_custom_fieldnames = [x for x in source_custom_fieldnames if x not in target_custom_fieldnames]
+for field in new_custom_fieldnames:
+    target_api.post('custom_fields/lead', data={'name': field, 'type': 'text'})
+    logging.info('added new custom field "%s"' % field)
+
+for lead in leads_to_transfer:
+    payload = {
+        'name': empty_if_none(lead['name']),
+        'url': empty_if_none(lead['url']),
+        'description': empty_if_none(lead['description']),
+        'status': status_map[lead['status_id']],
+        'addresses': lead['addresses'],
+        'custom': lead['custom'],
+        'date_created': lead['date_created'],
+        'date_updated': lead['date_updated']
+    }
+
+    new_contacts = []
+    for contact in lead['contacts']:
+        new_contacts.append({
+            'name': contact['name'],
+            'title': contact['title'],
+            'phones': [{'phone': x['phone'], 'type': x['type']} for x in contact['phones']],
+            'emails': [{'email': x['email'], 'type': x['type']} for x in contact['emails']],
+            'urls': [{'url': x['url'], 'type': x['type']} for x in contact['urls']],
+            'date_created': contact['date_created'],
+            'date_updated': contact['date_updated']
+        })
+    payload['contacts'] = new_contacts
+
+    if args.confirmed:
+        new_lead = target_api.post('lead', data=payload)
+    else:
+        new_lead['id'] = None
+        new_lead['name'] = lead['name']
+    logging.info('target: * added %s %s' % (new_lead['id'], new_lead['name']))
+
+    new_opportunities = []
+    for oppo in lead['opportunities']:
+        contact_ids = [x['id'] for x in new_lead['contacts'] if x['name'] == oppo['contact_name']]
+        contact_id = None
+        if contact_ids:
+            contact_id = contact_ids[0]
+
+        new_opportunities.append({
+            'lead_id': new_lead['id'],
+            'status': oppo['status_label'],
+            'value': oppo['value'],
+            'date_won': oppo['date_won'],
+            'date_lost': oppo['date_lost'],
+            'note': oppo['note'],
+            'value_period': oppo['value_period'],
+            'confidence': oppo['confidence'],
+            'contact_id': contact_id,
+            'date_created': oppo['date_created'],
+            'date_updated': oppo['date_updated']
+
+        })
+
+    for payload in new_opportunities:
+        if args.confirmed:
+            new_oppo = target_api.post('opportunity', data=payload)
+        else:
+            new_oppo['id'] = None
+        logging.info('target: %s added opportunity %s ' % (new_lead['id'], new_oppo['id']))
+
+    # tasks
+    new_tasks = []
+    for task in lead['tasks']:
+        new_tasks.append({
+            'lead_id': new_lead['id'],
+            'text': task['text'],
+            'due_date': task['due_date'],
+            'is_complete': task['is_complete']
+        })
```

Comment:
you need to copy over the `assigned_to` field (which is which user the task is assigned to)